### PR TITLE
fix: change month judgment logic

### LINKED ISSUE #95: [Bug] OpenLeaderBoard项目开源大屏每月大概1-3号大屏缺失内容
### Current Behavior

image

### Expected Behavior

希望解决方案可以避免当前缺失数据的情况，缺失数据比较影响用户体验。
基于这种情况是不是可以考虑传递的参数进行处理，避免出现大屏缺失数据的问题呢？

### Environment

```markdown
- OS:
- Browser:
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by andyhuang18**
## Brief Information

This pull request is in the type of ([more info](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#type) about types):

- [ ] build
- [ ] ci
- [ ] docs
- [ ] feat
- [x] fix
- [ ] perf
- [ ] refactor
- [ ] test

Related issues ([all available keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

- resolve #95 

## Details

目前修改的情况为下图所示。在调试环境中，我设定在6号之前，数据还处于未更新状态，因此当我选择 `2023-06` 时，红圈位置会显示提示信息。在我提交 Commit 的内容里，我设定在4号之前。目前未更新状态时会显示上上个月的数据。

shot

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/X-lab2017/open-leaderboard/blob/main/README.md) doc

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -52,6 +53,9 @@ const App = () => {
   if (lastUpdateTime) {
     [year, monthIndex] = getLastMonth(lastUpdateTime);
   }
+  if (CurrentDate.getDate() < 4) {
```
Comment (tyn1998) [inline src/index.js:56]: 魔力数字4跨文件出现了，可以考虑在一个地方export这个常量，然后在多处import使用。

File: src/components/table.js [hunk 2]
```diff
@@ -12,6 +12,8 @@ import expandObject from '../util/expandObject';
 import { t } from 'i18next';
 import './table.css';
 
+const updateDay = 4; //OpenDigger数据更新日期，每月4号更新
```
[the remark this answers is not in the document]
Comment (andyhuang18) [reply src/components/table.js:15]: 好的！

**Comment by tyn1998** [comment]
@andyhuang18 LGTM, thank you very much!

**Comment by tyn1998** [comment]
I should have run the code locally 😆 

directly pushed another [commit](https://github.com/X-lab2017/open-leaderboard/commit/e6b9ad6c391b4f9c6417b98104ff1f255f9b2ae7) to `main` to solve the problem.

**Comment by wj23027** [comment]
我有个小疑问，是open-leaderboard的所有数据都会延迟4天更新还是只是大屏延迟更新呢？

**Comment by andyhuang18** [comment]
> 我有个小疑问，是open-leaderboard的所有数据都会延迟4天更新还是只是大屏延迟更新呢？

都会延迟更新

**Pull request merged**